test/src/test/java/org/corfudb/runtime/view/stream/StreamAddressSpaceTest.java

```diff
@@ -40,7 +49,124 @@ public void testStreamAddressSpaceMerge() {
         assertThat(streamA.getTail()).isEqualTo(numStreamBEntries - 1);
 
         LongStream.range(streamBTrimMark + 1, numStreamBEntries).forEach(address ->
-                assertThat(streamA.getAddressMap().contains(address)).isTrue()
+                assertThat(streamA.contains(address)).isTrue()
         );
     }
+
+    @Test
+    public void constructorTest() {
+        StreamAddressSpace obj1 = new StreamAddressSpace(5L, Collections.emptySet());
+        assertThat(obj1.getTrimMark()).isEqualTo(5L);
+        assertThat(obj1.size()).isEqualTo(0L);
+
+        StreamAddressSpace obj2 = new StreamAddressSpace(5L, Collections.singleton(6L));
+        assertThat(obj2.getTrimMark()).isEqualTo(5L);
+        assertThat(obj2.size()).isEqualTo(1L);
+        assertThat(obj2.contains(6L)).isTrue();
+
+        StreamAddressSpace obj3 = new StreamAddressSpace(Collections.singleton(6L));
+        assertThat(obj3.getTrimMark()).isEqualTo(Address.NON_ADDRESS);
+        assertThat(obj3.size()).isEqualTo(1L);
+        assertThat(obj3.contains(6L)).isTrue();
+    }
+
+    @Test
+    @Ignore
```

Comment:
should this ignore be removed? I'm guessing it has to do with the commented code you had in the constructors...

Reply:
Fixed.